runtime/sema/checker.go

```diff
@@ -121,6 +121,9 @@ type Checker struct {
 	// memoryGauge is used for metering memory usage
 	memoryGauge  common.MemoryGauge
 	PositionInfo *PositionInfo
+
+	// Holds a mapping between references and their referenced-variable
+	references *VariableActivations
```

Comment:
Maybe compare this approach of implementing reference tracking with adding a new `referencedVariable *Variable` field to `Variable` and comparing the performance. We can also do this in a follow-up PR

Reply:
Seems there's no performance diff between the two. Anyway removed the bookkeeping since it makes the code simple bdde127d0906bb36a0f9dc051bf6b9a57020a229